### ISSUE TITLE
Wrong control-flow of break/continue inside block expressions

### ISSUE THREAD

**Issue by ortem** [intellij-rust/intellij-rust#6781]
## Environment

* **IntelliJ Rust plugin version:** 0.3.141.3638-203-nightly
* **Rust toolchain version:** 1.48.0 (7eac88abb 2020-11-16) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2020.3.2 Preview Ultimate Edition (IU-203.7148.15)
* **Operating system:** macOS 10.16
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
`break` and `continue` located inside block expressions break control-flow/data-flow analysis, e.g. liveness analysis:
```Rust
fn main() {
    let x = 1; // false-positive: Variable `x` is never used
    loop {
        {
            break;
        }
    }
    println!("{}", x);
}
```
